services/deployment-agent/src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -74,48 +96,93 @@ class GitRepo:  # pylint: disable=too-many-instance-attributes, too-many-argumen
     repo_id: str
     repo_url: URL
     branch: str
+    tags: str
     username: str
     password: str
     paths: List[Path]
     pull_only_files: bool
     directory: str = ""
 
+async def _checkout_repository(repo: GitRepo, tag: str = None):
+    if repo.pull_only_files:
+        await _git_checkout_files(repo.directory, repo.paths, tag)
+    else:
+        await _git_checkout_repo(repo.directory, tag)
 
-async def _init_repositories(repos: List[GitRepo]):
+async def _update_repository(repo: GitRepo):
+    if repo.pull_only_files:
+        await _git_pull_files(repo.directory, repo.paths)
+    else:
+        await _git_pull(repo.directory)
+
+async def _init_repositories(repos: List[GitRepo]) -> Dict:
+    description = {}
     for repo in repos:
         repo.directory = tempfile.TemporaryDirectory().name

```

Comment:
MINOR: cleanup [tempfile.TemporaryDirectory()](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) when done. Can also use a context manager with returned object

Reply:
well, this is running in a container and I though that in case of issue it might be better that the folder is still around for debugging.

Reply:
ok as soon as this is not constantly creating new folders ,right?